wp-includes/class-wp-https-ui.php

```diff
@@ -0,0 +1,220 @@
+		/* Translators: %s: a link for more details */
+		$insecure_content_description = esc_html__( 'Your home page doesn’t contain insecure URLs. However, there may be URLs on other pages that could be blocked. %s', 'pwa' );
+
+		/*
+		 * Todo: change !== to === as this is only like this for development.
+		 * The WP_HTTPS_Detection doesn't work with my local SSL certificate.
+		 * Also, change $this->is_currently_https() to ! $this->is_currently_https().
+		 * This is also for development only.
+		 * This main if block should only run if the site can use HTTPS, but the 'home' and 'siteurl' options aren't HTTPS.
+		 */
+		if ( $this->is_currently_https() && true !== get_option( WP_HTTPS_Detection::HTTPS_SUPPORT_OPTION_NAME ) ) :
+			?>
+			<p class="description">
+				<?php
+				echo wp_kses_post( sprintf(
+					/* Translators: %s: a link for more details */
+					esc_html__( 'HTTPS is essential to securing your WordPress site, we strongly suggest upgrading to HTTPS on your site. %s', 'pwa' ),
+					$https_more_details
+				) );
+				?>
+			</p>
+			<p style="margin-top: 20px;"><strong><?php esc_html_e( 'HTTPS Upgrade', 'pwa' ); ?></strong></p>
+			<p>
+				<label><input name="<?php echo esc_attr( self::UPGRADE_HTTPS_OPTION ); ?>" type="radio" <?php checked( $upgrade_https_value, self::OPTION_SELECTED_VALUE ); ?> value="<?php echo esc_attr( self::OPTION_SELECTED_VALUE ); ?>"><?php esc_html_e( 'Yes', 'pwa' ); ?></label>
+				<label><input name="<?php echo esc_attr( self::UPGRADE_HTTPS_OPTION ); ?>" type="radio" <?php checked( self::OPTION_SELECTED_VALUE !== $upgrade_https_value ); ?> value="0" ><?php esc_html_e( 'No', 'pwa' ); ?></label>
+			</p>
+			<p class="description"><?php esc_html_e( 'Your site appears to support HTTPS', 'pwa' ); ?></p>
+
+			<p style="margin-top: 20px;"><strong><?php esc_html_e( 'Upgrade Insecure URLs', 'pwa' ); ?></strong></p>
+			<p>
+				<label><input name="<?php echo esc_attr( self::UPGRADE_INSECURE_CONTENT_OPTION ); ?>" type="radio" <?php checked( $upgrade_insecure_content, self::OPTION_SELECTED_VALUE ); ?> value="<?php echo esc_attr( self::OPTION_SELECTED_VALUE ); ?>"><?php esc_html_e( 'Yes', 'pwa' ); ?></label>
+				<label><input name="<?php echo esc_attr( self::UPGRADE_INSECURE_CONTENT_OPTION ); ?>" type="radio" <?php checked( self::OPTION_SELECTED_VALUE !== $upgrade_insecure_content ); ?> value="0" ><?php esc_html_e( 'No', 'pwa' ); ?></label>
+			</p>
+			<p class="description">
+				<?php echo wp_kses_post( sprintf( $insecure_content_description, $insecure_content_more_details ) ); ?>
+			<p>
+			<?php
+		else :
+			/* Translators: %s: HTTPS more details link */
+			echo wp_kses_post( sprintf( __( 'Your site doesn\'t look to support HTTPS. %s', 'pwa' ), $https_more_details ) );
+		endif;
+	}
+
+	/**
+	 * Whether the options indicate that the site is currently using HTTPS.
+	 *
+	 * @return boolean
+	 */
+	public function is_currently_https() {
+		$urls = array( get_option( 'home' ), get_option( 'siteurl' ) );
```

Comment:
Instead of using the underlying `get_option()` calls it would be better to use `home_url()` and `site_url()` since additional filters apply to their return values which need to be accounted for. That is, unless the goal is to get the underlying value stored in the DB.

Reply:
Sure, [https://github.com/xwp/pwa-wp/pull/36/commits/2b9eceb5fb1778521cf6e9d16265cebff36af0a5](https://github.com/xwp/pwa-wp/pull/36/commits/2b9eceb5fb1778521cf6e9d16265cebff36af0a5) uses `home_url()` and `site_url()` instead of the `get_option()` calls.